### ISSUE TITLE
alt attribute on lazy-loaded <picture> tags

### ISSUE THREAD

**Issue by 0gust1** [ApoorvSaxena/lozad.js#92]
For the sake of accessibility, it would be cleaner if lazy-loaded images using `<picture>`tags could "inherit" their `alt`attribute from a data-X attribute on picture tag.

On our project we easily managed that by overloading the `load`function of lozad (and using `data-lazy-img-alt` attribute). It could be nice if it could be default (a11y is important).

There can be two ways : 
* make this functionnality by default
* write a documentation paragraph explaining how to overload the `load`function.

Attempting to do a pull-request in this direction, I was confronted to the problem of test coverage (see #93)

**Comment by ApoorvSaxena** [comment]
@0gust1 if test case addition in their current form is hard to implement, then we can move ahead atleast with the functionality addition in the library followed by test cases.

**Comment by robots4life** [comment]
I am using picture tags with the img alt attribute for a client that needs a11y.
How can I use lozad while still making sure that in the end the img alt attribute is correctly shown?
Is there a step to step procedure I could follow to achieve this?

Or how does this 
> overloading the loadfunction of lozad and using data-lazy-img-alt attribute

work please?

**Comment by 0gust1** [comment]
@robots4life : 
I should have said `override` instead of `overload` :)

Basically, you can easily customize the behavior of lozad, and use your own loading function (code from documentation) :

```javascript
lozad('.lozad', {
    load: function(el) {
        console.log('loading element');

        // Custom implementation to load an element
        // e.g. el.src = el.getAttribute('data-src');
    }
});
```

What we have done in our project : we made our own loading function, based on the default one of lozad : 

```javascript
function loadFun(el) {
  const isIE = document.documentMode;

  if (el.nodeName.toLowerCase() === 'picture') {
    const imgTag = document.createElement('img');

    //IE management
    if (isIE && el.getAttribute('data-iesrc')) {
      imgTag.src = el.getAttribute('data-iesrc');
    }

    // apply alt attribute to the image if corresponding data attribute
    // is found on the <picture> tag
    if (el.getAttribute('data-lazy-img-alt')) {
      imgTag.alt = el.getAttribute('data-lazy-img-alt');
    }

    // apply css class to the image if corresponding data attribute
    // is found on the <picture> tag
    if (el.getAttribute('data-lazy-img-class')) {
      imgTag.className = el.getAttribute('data-lazy-img-class'); // we use classname because the attribute could contain several classnames
    }

    // NRT management
    if (el.getAttribute('data-tnr')) {
      imgTag.dataset = el.getAttribute('data-tnr');
    }

    el.appendChild(imgTag);
  }

  // <img> and <iframe> case
  if (el.getAttribute('data-src')) {
    el.src = el.getAttribute('data-src');
  }

  // <img>
  if (el.getAttribute('data-srcset')) {
    el.srcset = el.getAttribute('data-srcset');
  }

  // CSS background case
  if (el.getAttribute('data-background-image')) {
    el.style.backgroundImage = `url(${el.getAttribute(
      'data-background-image'
    )})`;
  }
}
```

**Comment by robots4life** [comment]
@0gust1 Thank you heaps for the detailed explanation. Invaluable help! Will try to get this going and report back.

**Comment by dooman87** [comment]
Hey guys, I raised a PR for this feature. Appreciate if you can have a look.

**Comment by ApoorvSaxena** [comment]
feature has been released in v1.5.0 of Lozad and added in README as well, thanks everyone for their contributions on this thread
